features/admin.users.v1/components/user-groups-list.tsx

```diff
@@ -0,0 +1,493 @@
+import { updateGroupDetails } from "@wso2is/admin.groups.v1/api/groups";
+import { GroupsMemberInterface, PatchGroupDataInterface } from "@wso2is/admin.groups.v1/models/groups";
+import {
+    AlertLevels,
+    IdentifiableComponentInterface,
+    ProfileInfoInterface
+} from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import {
+    ConfirmationModal,
+    DataTable,
+    EmphasizedSegment,
+    EmptyPlaceholder,
+    Heading,
+    LinkButton,
+    ListLayout,
+    PrimaryButton,
+    TableActionsInterface,
+    TableColumnInterface,
+    UserAvatar
+} from "@wso2is/react-components";
+import isEmpty from "lodash-es/isEmpty";
+import React, {
+    FunctionComponent,
+    ReactElement,
+    ReactNode,
+    SyntheticEvent,
+    useEffect,
+    useState
+} from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider, DropdownProps, Header, Icon, Label, PaginationProps, SemanticICONS } from "semantic-ui-react";
+import "./user-roles-list.scss";
+
+interface UserGroupsEditPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * User profile
+     */
+    user: ProfileInfoInterface;
+    /**
+     * Is the component loading
+     */
+    isLoading?: boolean;
+    /**
+     * Is the component in read only mode
+     */
+    isReadOnly?: boolean;
+    /**
+     * Callback to open the add new group modal
+     */
+    handleOpenAddNewGroupModal: () => void;
+    /**
+     * Handle user update callback.
+     */
+    handleUserUpdate: (userId: string) => void;
+}
+
+export const UserGroupsListTable: FunctionComponent<UserGroupsEditPropsInterface> = (
+    props: UserGroupsEditPropsInterface
+): ReactElement => {
+
+    const {
+        user,
+        isLoading,
+        isReadOnly,
+        handleOpenAddNewGroupModal,
+        handleUserUpdate,
+        "data-componentid": componentId
+    } = props;
+
+    const { t } = useTranslation();
+
+    const dispatch: Dispatch = useDispatch();
+
+    const [ searchQuery, setSearchQuery ] = useState<string>("");
+    const [ listItemLimit, setListItemLimit ] = useState<number>(UIConstants.DEFAULT_RESOURCE_LIST_ITEM_LIMIT);
+    const [ listOffset, setListOffset ] = useState<number>(0);
+    const [ paginatedGroups, setPaginatedGroups ] = useState<GroupsMemberInterface[]>(undefined);
+    const [ triggerClearQuery, setTriggerClearQuery ] = useState<boolean>(false);
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ deletingGroup, setDeletingGroup ] = useState<GroupsMemberInterface>(undefined);
+    const [ isDeleting, setDeleting ] = useState<boolean>(false);
+
+    const userGroupsList: GroupsMemberInterface[] = user?.groups;
+
+    /**
+     * Set initial selected roles options
+     */
+    useEffect(() => {
+        setGroupsPage(0, listItemLimit, userGroupsList, searchQuery);
```

Comment:
the function name looks like a setter of a react state variable. shall we update the function name?

Reply:
Fixed with cb0ee2cf165319ca096aa44d358799060f08bbf9